mypy/checkexpr.py

```diff
@@ -1439,6 +1439,23 @@ def check_for_extra_actual_arguments(self,
                                                                             context)
                                 is_unexpected_arg_error = True
                         ok = False
+                elif isinstance(actual_type, Instance) and \
+                        actual_type.type.has_base('typing.Mapping'):
+                    if messages and actual_type.args:
```

Comment:
The type does not necessarily have the same args as `Mapping`. For example, we could have a class like this:

```py
class C(Mapping[str, T]): ...
```

To make this work in general, you can use `map_instance_to_supertype`.